crates/primitives/src/transaction/mod.rs

```diff
@@ -727,42 +692,50 @@ impl TransactionSigned {
     /// Inner encoding function that is used for both rlp [`Encodable`] trait and for calculating
     /// hash that for eip2728 does not require rlp header
     pub(crate) fn encode_inner(&self, out: &mut dyn bytes::BufMut, with_header: bool) {
-        if let Transaction::Legacy(TxLegacy { chain_id, .. }) = self.transaction {
-            let header = Header { list: true, payload_length: self.payload_len() };
-            header.encode(out);
-            self.transaction.encode_fields(out);
-
-            if let Some(id) = chain_id {
-                self.signature.encode_eip155_inner(out, id);
-            } else {
-                // if the transaction has no chain id then it is a pre-EIP-155 transaction
-                self.signature.encode_inner_legacy(out);
-            }
-        } else {
-            if with_header {
-                let header = Header { list: false, payload_length: self.payload_len() };
+        match self.transaction {
+            Transaction::Legacy(TxLegacy { chain_id, .. }) => {
+                // do nothing w/ with_header
```

Comment:
what does `with_header` refer to?

Reply:
Please note the document on the function, 

```
/// Inner encoding function that is used for both rlp [`Encodable`] trait and for calculating
/// hash that for eip2728 does not require rlp header
```

`with_header` is an function argument that already exists. When I first read this function, I think it's little bit wierd because, For `Legacy` transaction, `with_header` argument is not used at all. It's only used for `Non` legacy transactions, EIP2930 and EIP1559. I tried to remove `with_header` argument from this function but I got to know it's needed for EIP2778 from the function document. So I'd like to leave a note explicitly to note that `with_header` argument is not used for `Legacy` transaction.

Reply:
If you think this comment is not proper, I'll remove this comment. :)

Reply:
> do nothing w/ with_header

some context here: 
the header argument only applies for typed tx which are enveloped with the id. however <id + rlp(tx)> is itself not rlp so in order to convert this to rlp you need to prepend it with the header for <id +rlp(tx)>

however, you don't need this for calculating the receipt root for example

So I'd suggest the following:
* update the comment
* convert bool `with_header` argument into an `enum` maybe `EncodeTarget {Rlp, Raw}`